File: cesta/IvanRC-Cesta/IvanRC-Cesta-Java/src/main/java/FactoriesCesta/FactoryUsuario.java
package FactoriesCesta;

import ModelsCesta.Usuario;

public class FactoryUsuario {

    public FactoryUsuario(){}
    private static FactoryUsuario instance = null;

    public static FactoryUsuario getInstance(){
        if(instance == null){
            instance = new FactoryUsuario();
        }
        return instance;
    }
    private static int contadorUsuarios = 1;
    private int nextId(){
        return contadorUsuarios++;
    }

    /**
     * función que sirve para crear un usuario aleatorio
     * @return el usuario cuyas propiedades se crearon al azar
     */
    public Usuario crearUsuarioRandom(){
        String[] nombres = {"Jose", "Paco", "Manuel", "Iván", "Romeo", "Belén", "Teseo"};
        return new Usuario(nextId(), nombres[(int) (Math.random()* nombres.length)]);
    }
}

File: cesta/IvanRC-Cesta/IvanRC-Cesta-Kotlin/src/main/kotlin/FactoriesCestaDeCompra/FactoryUsuario.kt
package FactoriesCestaDeCompra

import ModelsCestaDeCompra.Usuario

class FactoryUsuario {

    companion object{
        private var instance: FactoryUsuario? = null

        fun getInstance(): FactoryUsuario?{
            if(instance == null){
                instance = FactoryUsuario()
            }
            return instance
        }
        var contadorUsuarios = 1
        fun nextId():Int{
            return contadorUsuarios++
        }
    }

    /**
     * función que sirve para crear un usuario aleatorio
     * @return el usuario cuyas propiedades se crearon al azar
     */
    fun crearUsuarioRandom(): Usuario{
        val nombres = arrayOf("Jose", "Paco", "Manuel", "Iván", "Romeo", "Belén", "Teseo")
        return Usuario(nextId(), nombres.random())
    }
}